src/Stack/SourceMap.hs

```diff
@@ -111,26 +113,26 @@ getPLIVersion (PLIRepo _ pm) = pkgVersion $ pmIdent pm
 globalsFromDump ::
        (HasLogFunc env, HasProcessContext env)
     => ActualCompiler
-    -> RIO env (Map PackageName GlobalPackage)
+    -> RIO env (Map PackageName Version)
 globalsFromDump compiler = do
     let pkgConduit =
             conduitDumpPackage .|
             CL.foldMap (\dp -> Map.singleton (dpGhcPkgId dp) dp)
         toGlobals ds = Map.fromList $ map toGlobal $ Map.elems ds
         toGlobal d =
             ( pkgName $ dpPackageIdent d
-            , GlobalPackage (pkgVersion $ dpPackageIdent d))
+            , pkgVersion $ dpPackageIdent d)
     toGlobals <$> ghcPkgDump (whichCompiler compiler) [] pkgConduit
 
 globalsFromHints ::
        HasConfig env
     => WantedCompiler
-    -> RIO env (Map PackageName GlobalPackage)
+    -> RIO env (Map PackageName Version)
```

Comment:
I'm almost tempted to say we should have a special `GlobalVersion` newtype wrapper to drive the point home further, but this is good.